edb/server/protocol/binary.pyx

```diff
@@ -606,20 +608,23 @@ cdef class EdgeConnection(frontend.FrontendConnection):
         if user not in roles:
             raise errors.AuthenticationError('authentication failed')
 
-    def _auth_jwt(self, user):
+    def _auth_jwt(self, user, encoded_token):
         role = self.server.get_roles().get(user)
         if role is None:
             raise errors.AuthenticationError('authentication failed')
 
-        if not self._auth_data:
-            raise errors.AuthenticationError(
-                'authentication failed: no authorization data provided')
+        decrypt = False
+        if not encoded_token:
+            decrypt = True  # token in binary protocol over HTTP is encrypted
```

Comment:
Let's just drop encryption of JWT and normalize this (i.e. check for `token` in `params` and then in `self._auth_data`)